Sources/SpeziFHIRInterpretation/FHIRInterpretationModule.swift

```diff
@@ -0,0 +1,99 @@
+//
+// This source file is part of the Stanford LLM on FHIR project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Spezi
+import SpeziFHIR
+import SpeziLLM
+import SpeziLLMOpenAI
+import SpeziLocalStorage
+import SwiftUI
+
+
+public class FHIRInterpretationModule: Module, DefaultInitializable {
+    public enum Defaults {
+        public static var llmSchema: LLMOpenAISchema {
+            .init(
+                parameters: .init(
+                    modelType: .gpt4_turbo_preview,
+                    systemPrompt: nil   // No system prompt as this will be determined later by the resource interpreter
+                )
+            )
+        }
+    }
+    
+    
+    @Dependency private var localStorage: LocalStorage
+    @Dependency private var llmRunner: LLMRunner
+    @Dependency private var fhirStore: FHIRStore
+    
+    @Model private var resourceSummary: FHIRResourceSummary
+    @Model private var resourceInterpreter: FHIRResourceInterpreter
+    @Model private var multipleResourceInterpreter: FHIRMultipleResourceInterpreter
+    
+    // TODO: Adjust this to make it configurable after the fact
+    // UserDefaults.standard.string(forKey: StorageKeys.openAIModel) ?? StorageKeys.Defaults.openAIModel
+    let summaryLLMSchema: any LLMSchema
+    let interpretationLLMSchema: any LLMSchema
+    let openAIModelType: LLMOpenAIModelType
+    
+    
+    public func configure() {
+        resourceSummary = FHIRResourceSummary(
+            localStorage: localStorage,
+            llmRunner: llmRunner,
+            llmSchema: summaryLLMSchema
+        )
+        
+        resourceInterpreter = FHIRResourceInterpreter(
+            localStorage: localStorage,
+            llmRunner: llmRunner,
+            llmSchema: interpretationLLMSchema
+        )
+        
+        multipleResourceInterpreter = FHIRMultipleResourceInterpreter(
+            localStorage: localStorage,
+            llmRunner: llmRunner,
+            llmSchema: LLMOpenAISchema(
+                parameters: .init(
+                    modelType: openAIModelType,
+                    systemPrompt: nil   // No system prompt as this will be determined later by the resource interpreter
+                )
+            ) {
+                // FHIR interpretation function
+                FHIRGetResourceLLMFunction(
+                    fhirStore: self.fhirStore,
+                    resourceSummary: self.resourceSummary,
+                    allResourcesFunctionCallIdentifier: self.fhirStore.allResourcesFunctionCallIdentifier
+                )
```

Comment:
One thing that I forgot in the meeting today: An element that we capture here are the `allResourcesFunctionCallIdentifier`s. In case, e.g., the FHIR Store updates we won't update the function parameters.

Doesn't have to be solved in this PR but an eventual solution would be to use the `Observable` conformance of `FHIRStore` and utilize this to get informed about updates, re-assign the parameter property wrapper with new values, and then re-register for observation changes.

Probably means that the `allResourcesFunctionCallIdentifier` would be come optional to allows a function to derive them from the fhirstore. Maybe we should rename it to `allowedResourcesFunctionCallIdentifiers`? Might be that a user could pass in a subset of identifiers if they want.